### PR TITLE
Add serde support

### LINKED ISSUE #5: Add serde support feature
Hello,
I think serde support would be nice.

### PR DESCRIPTION
**Pull request by IcanDivideBy0**
fixes #5

### CODE REVIEW

**Comment by zbrox** [comment]
Hey, just a couple of comments on this.

Can you add the `features` section in `Cargo.toml` with an empty default and all. We should also rename the `serde` feature to something like `with_serde` or something like that. The cargo docs advise that feature names don't conflict with packages because the features list in Cargo.toml has a single namespace. And also can you add some simple tests for serializing and deserializing.

**Comment by zbrox** [comment]
Oh, I also now noticed the branch. Can you change it to the develop branch.

**Comment by IcanDivideBy0** [comment]
@zbrox I added the `with-serde` feature flag in `Cargo.toml`.
> Can you change it to the develop branch.
What do you mean by that ? Do you want me to rebase on the develop branch ?

**Comment by IcanDivideBy0** [comment]
Oh ok, I see you already fixed clippy warnings in the develop branch, it's rebased!

File: src/lib.rs [hunk 1]
```diff
@@ -1502,15 +1507,16 @@ impl Currency {
...
-    /// 
+    ///
     /// ```
     /// use iso_currency::Currency;
-    /// 
+    ///
     /// assert_eq!(Currency::from_numeric(978), Some(Currency::EUR));
     /// ```
     pub fn from_numeric(numeric_code: u16) -> Option<Currency> {
+        #[allow(clippy::zero_prefixed_literal)]
```
Comment (IcanDivideBy0) [inline src/lib.rs:1519]: You already added it to the crate root, but IMO it's better to disable lints on a restricted scope?
Do you want me to remove this or the root one?

Comment (zbrox) [reply src/lib.rs:1519]: Yeah, you're right let's remove the root level one.

**Comment by zbrox** [comment]
> @zbrox I added the `with-serde` feature flag in `Cargo.toml`.
> 
> > Can you change it to the develop branch.
> > What do you mean by that ? Do you want me to rebase on the develop branch ?

Yeah, sorry about that. In the haste I was a bit unclear. I meant to rebase it to develop and to change the base branch of the pull request to develop. But to my dismay I can edit the base branch myself :) I've missed quite a lot in the recent years in GitHub UI development it seems. 

It's just that I keep the current work on develop and merge to master only for releases.

File: src/lib.rs [hunk 2]
```diff
@@ -21,7 +21,11 @@
 //! assert_eq!(format!("{}", Currency::EUR.symbol()), "€");
 //! ```
 
-#[derive(PartialEq)]
+#[cfg(feature = "serde")]
```
Comment (zbrox) [inline src/lib.rs:24]: Just need to change to the new feature name.

Comment (IcanDivideBy0) [reply src/lib.rs:24]: oops

**Comment by zbrox** [comment]
Thanks for the contribution. I can tomorrow add some simple serialization/deserialization tests.

**Pull request merged**